core/src/main/java/feign/template/BodyTemplate.java

```diff
@@ -35,10 +41,24 @@ public static BodyTemplate create(String template) {
 
   private BodyTemplate(String value, Charset charset) {
     super(value, ExpansionOptions.ALLOW_UNRESOLVED, EncodingOptions.NOT_REQUIRED, false, charset);
+    if (value.startsWith(JSON_TOKEN_START_ENCODED) && value.endsWith(JSON_TOKEN_END_ENCODED)) {
+      this.json = true;
+    }
   }
 
   @Override
   public String expand(Map<String, ?> variables) {
-    return UriUtils.decode(super.expand(variables), Util.UTF_8);
+    String expanded = super.expand(variables);
+    if (this.json) {
+      /* decode only the first and last character */
+      expanded = expanded.substring(
+          expanded.indexOf(JSON_TOKEN_START_ENCODED) + JSON_TOKEN_START_ENCODED.length());
+      expanded = JSON_TOKEN_START + expanded;
+      expanded = expanded.substring(0, expanded.lastIndexOf(JSON_TOKEN_END_ENCODED));
+      expanded += JSON_TOKEN_END;
```

Comment:
Shouldn't we be using a `StringBuilder` ?

Reply:
Yes, that would be better.  I'll make the changes.